doc/beamline_completion.md

```diff
@@ -71,25 +71,84 @@ This is a breaking change so would have to happen at some controlled point.
+
+Here are some key differences as relating to the Geometry:
+
+* Fundamental non-compatibility between `Workspace` and `Dataset`. `Dataset` has more information than possible in `Workspace`. All interop via conversion, API compatibility not maintained.
+  - Existing saving would have to "slice" `Dataset`, so would be required to write new saving functionality and format for `Dataset`
+  - Because of saving requirement, no requirement to save geometry deltas. Geometry can be written to be directly correspond to the mutated Beamline geometry - not origninal (IDF) + delta
+  - Loading functionality would follow from the above
+* No ExperimentInfo
+  - Geometry, alongside run information is part of `Dataset`
+* No ParameterMap
+  - Exact solution(s) not yet apparent, but there are several options. Given that a number of parameters are linked to the instrument and not individual components, this might affect how things are done
+* No `ComponentID` by design. Infact, there may be no requirement for unique identification of `Components` at all.
+
+Another key difference with the phases for non-dataset approach is that the approach concludes with the complete removal of `Instrument 1.0` but not `Workspace 1.0`. The current preferred rollout for `Dataset` is to develop alongside the existing codebase, and not to consider removal of legacy data structures i.e. Workspace until `Dataset` usage has acted as a full replacement. We would simultaneously remove `Workspace 1.0`, `Algorithms 1.0`, `Instrument 1.0`, `ExperimentInfo`. 
+
+The net result of these differences is that the activities and timescales differ from those presented in the above section. We refer to these as activities rather than phases, since the order of the activities is not known and possibly not interdependent.
+
+### Activity 1 - Functional wrappers to Dataset for Geometry
+
+Need to provide classes of the form `SpectrumInfo`, `ComponentInfo` and `DetectorInfo` that wrap `Dataset`. These classes should point to a `Dataset` containing all required information. Note that `ComponentInfo` and `DetectorInfo` are inter-dependent, a higher-level grouping may be required. `SpectrumInfo` uses `DetectorInfo`. We would need to consider time spent to reproduce functional and performance testing.
+
+### Activity 2 - Saving Loading
+
+As described above, no need to preserve compatibility with existing intermediate formats as far as the total `Dataset` is concerned. Time would need to be invested to trial and optimise these. Requirements for end-users should also be considered, for example if `Dataset` libraries are used outside of Mantid. Note that `Saving` and `Loading` and format specifics must not be mixed into the `Dataset` library. 
```

Comment:
Yes - interesting point, we could adopt the Nexus approach for storing shapes.

Reply:
Should work well with dataset. Just meshes and special case for cylinder.

Reply:
@SimonHeybrock are you aware for any need for constructive solid geometry in future instruments. If not, I suggest we implement shapes as above. Again, `ComponentInfo` class would be useful to provide usable shapes.

Reply:
I cannot tell, since I do not know what it is used for right now. Note that it may also be required for the sample anyway?

Reply:
I think this is the thing to do. Would guarantee to work for ESS since there will be no possibility for CSG in the geometry specification of raw files. Doing anything else would be a big headache - not sure how a CSG specification would map to `dataset` well.